### PR TITLE
add tests and fix index overflow on Atlassian Fork

### PR DESCRIPTION
**Pull request by Tenaria**
**Description:** <Describe what has changed.>
This fixes the bug outlined [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7250) by ensuring that we are comparing float64s. Since the time.Duration value is represented by an int64 in the underlying implementation, this means we'll always need to append the `math.maxFloat64` value as the "infinity bucket" no matter what the user specifies.

In theory we shouldn't encounter this bug since it requires a span with a very large timestamp but it'll be good to fix it to be safe.

**Testing:** Adds in tests that would catch that case and throw an index out of bounds error

**Documentation:** N/A - internal fix

### CODE REVIEW

File: processor/spanmetricsprocessor/processor_test.go [hunk 1]
```diff
@@ -333,10 +373,13 @@ func TestProcessorConsumeTraces(t *testing.T) {
 			mexp := &mocks.MetricsExporter{}
 			tcon := &mocks.TracesConsumer{}
 
-			//TODO: CLAIRE: CHECK IF WE CAN ADD TO THESE TESTS
 			// Mocked metric exporter will perform validation on metrics, during p.ConsumeTraces()
 			mexp.On("ConsumeMetrics", mock.Anything, mock.MatchedBy(func(input pdata.Metrics) bool {
-				return tc.verifier(t, input, false, make(map[string]int))
+				if tc.verifier != nil {
```
Comment (jamesmoessis) [inline processor/spanmetricsprocessor/processor_test.go:378]: Why would it be ok for the verifier to be `nil`? This is what validates the metrics. 

Also, even if you don't need the verifier for a test case, I would probably just set it as an empty function `func(...) {}` in the test case, rather than checking if it's nil for every test case.

Comment (Tenaria) [reply processor/spanmetricsprocessor/processor_test.go:378]: The main purpose of the test was to check that it didn't cause an out of bounds index error rather than verifying the correct metrics were generated so that's why I set it to nil. 

I guess it couldn't hurt to verify anyway but the existing metric verifier also wasn't written generically enough so that I could easily use it for the specific use case and I didn't think it was particularly worth rewriting given that we are planning on removing the aggregation which will simplify things and our existing tests will still give us enough confidence and testing.

I'll do the suggestion for setting the empty function 👍

**Comment by Tenaria** [comment]
bump, reminder for reviews on this PR please and thanks!

File: processor/spanmetricsprocessor/processor_test.go [hunk 2]
```diff
@@ -323,6 +361,12 @@ func TestProcessorConsumeTraces(t *testing.T) {
 			verifier:               verifyConsumeMetricsInputDelta,
 			traces:                 []pdata.Traces{buildSampleTrace(), buildSampleTrace()},
 		},
+		{
+			name:                   "Test maximum span time will not cause out of bounds index error",
+			aggregationTemporality: delta,
+			verifier:               verifyMetricsNoOp,
```
Comment (jamesmoessis) [inline processor/spanmetricsprocessor/processor_test.go:367]: probably better to use an inline, anonymous function here

**Review by jamesmoessis — APPROVED** [review]

File: processor/spanmetricsprocessor/processor_test.go (hunk 2 above)
Comment (Tenaria) [reply processor/spanmetricsprocessor/processor_test.go]: fixed
